core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -452,12 +452,13 @@ class ZkUtils(val zkClient: ZkClient,
     } catch {
       case e1: ZkBadVersionException =>
         optionalChecker match {
-          case Some(checker) => return checker(this, path, data)
-          case _ => debug("Checker method is not passed skipping zkData match")
+          case Some(checker) => checker(this, path, data)
+          case _ =>
+            debug("Checker method is not passed skipping zkData match")
+            warn("Conditional update of path %s with data %s and expected version %d failed due to %s"
+              .format(path, data,expectVersion, e1.getMessage))
+            (false, -1)
```

Comment:
Yeah, its a bit unfortunate. It looks like its a Scala issue tracked by [SI-8979](https://issues.scala-lang.org/browse/SI-8979). More discussion on the Scoverage issue can be seen [here](https://github.com/scoverage/scalac-scoverage-plugin/issues/93).